index.js

```diff
@@ -62,7 +86,106 @@ function createTaskCreatedAt(taskCreatedAt) {
 function createTaskCheckBox() {
   return `<input
             type="checkbox"
-            class="table-box-check-box"
             name="checkbox-done"
           /> `;
 }
+
+// function sortTasks(sortType) {
+//   if (sortType == "byTitle") {
+//     renderTasks(sortTasksByTitle());
+//   } else {
+//     renderTasks(unsortTasks());
+//   }
+// }
+
+// function sortTasksByTitle() {
+//   const collator = new Intl.Collator("en", {
+//     numeric: true,
+//   });
+
+//   return tasks.sort((randomTask1, randomTask2) =>
+//     collator.compare(
+//       randomTask1.title.toUpperCase(),
+//       randomTask2.title.toUpperCase()
+//     )
+//   );
+// }
+
+// function unsortTasks() {
+//   return tasks.sort((task1, task2) => (task1.id > task2.id ? 1 : -1));
+// }
+
+function filterTasksByTitle(text) {
+  renderTasks(tasks.filter((task) => task.title.includes(text)));
+}
+
+function filterTasksByTime(time) {
+  if (time == "") {
+    renderTasks(tasks);
+  }
+  if (time == "today") {
+    renderTasks(filterTodayTasks());
+  } else if (time == "last-day") {
+    renderTasks(filterLastDayTasks());
+  } else if (time == "last-7-days") {
+    renderTasks(filterLast7DaysTasks());
+  } else if (time == "last-30-days") {
+    renderTasks(filterLast30DaysTasks());
+  }
+}
+
+function filterTodayTasks() {
+  let filteredTasks = [];
+
+  tasks.forEach((task) => {
+    const taskTime = new Date(task.createdAt).setHours(0, 0, 0, 0);
+    const today = new Date().setHours(0, 0, 0, 0);
+
+    if (taskTime == today) {
+      filteredTasks.push(task);
+    }
+  });
+  return filteredTasks;
```

Comment:
- Can be done with `.map` instead of `.forEach`. Then you don't have to define `filteredTasks`.
- You don't have to `setTime()` of the `taskTime` itself. You can just use `taskTime > today` (sudo code).

Reply:
Please apply this for the rest of functions below.